oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -109,6 +109,40 @@ public class ComputeEngineCredentials extends GoogleCredentials
   static final int MAX_COMPUTE_PING_TRIES = 3;
   static final int COMPUTE_PING_CONNECTION_TIMEOUT_MS = 500;
 
+  public enum AuthTransport {
```

Comment:
I think AuthTransport is ok for this. I would prefer if this could be more specific if possible.

Is there a term for alts and mtls type of transport that we can use? Does `TlsTransport` make sense? I don't know if TLS has a transport concept or if that would be confusing.

`GoogleMtlsTransport`?

Reply:
TLDR: WDYT of `GoogleAuthTransport`?

> I think AuthTransport is ok for this. I would prefer if this could be more specific if possible.

We generally call this "transport type": describes how the client is authenticating to Google APIS either via GFE (using `mtls`) or via DirectPath (using `alts`). However Blake mentioned above that transport has a different meaning in Client libraries (grpc vs http), so it would be good to clarify to something more specific, which is how we landed on `AuthTransport` initially.

> Is there a term for alts and mtls type of transport that we can use? Does TlsTransport make sense? I don't know if TLS has a transport concept or if that would be confusing.

TLS is the security protocol being used, it's used in both mtls and alts. mtls is mutually authenticated TLS, alts is (in short) "Google-optimized" mtls (see [public docs](https://cloud.google.com/docs/security/encryption-in-transit/application-layer-transport-security)).

In the enum name, I think we want to focus on how clients are authenticating to Google APIS, not details of the protocol being used (mtls vs alts). How about `GoogleAuthTransport`? I think this captures that the enum is listing different ways to ways clients can authenticate to Google APIs.

> GoogleMtlsTransport? Though I'm not sure mtls/atls are google specific terms.
I think this name would be ok. One thing is that mtls via GFE strictly adheres to the TLS standard (i.e. not Google specific). Whereas alts is a [Google specific protocol](https://cloud.google.com/docs/security/encryption-in-transit/application-layer-transport-security).

I've done this in https://github.com/googleapis/google-auth-library-java/pull/1558/commits/8ce3a4d6782d58b13a90fbf7899b28a02419adf6, let me know if the name sounds ok.